tools/run_tests/helper_scripts/build_csharp.sh

```diff
@@ -32,5 +32,12 @@ set -ex
 
 cd $(dirname $0)/../../../src/csharp
 
-# overriding NativeDependenciesConfigurationUnix is needed to make gcov code coverage work.
-xbuild /p:Configuration=$MSBUILD_CONFIG /p:NativeDependenciesConfigurationUnix=$CONFIG Grpc.sln
+OVERRIDE_NATIVELIB_MAYBE=""
+if [ "$CONFIG" == "gcov" ]
+then
+  # overriding NativeDependenciesConfigurationUnix makes C# project pick up
+  # the gcov flavor of grpc_csharp_ext
+  OVERRIDE_NATIVELIB_MAYBE="/p:NativeDependenciesConfigurationUnix=$CONFIG"
```

Comment:
I think it would be nice if gcov wasn't a special case here, but guessing that NativeDependenciesConfigurationUnix=$CONFIG" breaks down elsewhere?

otherwise not totally necessary tweak but maybe:

```
if [ "$CONFIG" == "gcov" ]
then
  dotnet build --configuration $MSBUILD_CONFIG /p:NativeDependenciesConfigurationUnix=gcov Grpc.sln
else
  dotnet build --configuration $MSBUILD_CONFIG Grpc.sln
fi
```

?